io_scene_godot/converters/material_node_tree.py

```diff
@@ -0,0 +1,857 @@
+    two output nodes found, raise error"""
+    output_node = None
+    for node in node_tree.nodes:
+        if node.bl_idname == 'ShaderNodeOutputMaterial':
+            if output_node is None:
+                output_node = node
+            else:
+                raise ValidationError(
+                    "More than one Material Ouput node found")
+    return output_node
+
+
+def export_texture(escn_file, export_settings, image):
+    """Export texture image as an external resource"""
+    resource_id = escn_file.get_external_resource(image)
+    if resource_id is not None:
+        return resource_id
+
+    dir_path = os.path.dirname(export_settings['path'])
+    image_extension = '.' + image.file_format.lower()
+    image_path = dir_path + os.sep + image.name + image_extension
+    image.filepath_raw = image_path
+    image.use_alpha = True
+    image.alpha_mode = 'STRAIGHT'
+    image.file_format = image.file_format
+    image.save()
+
+    img_resource = ExternalResource(image_path, "Texture")
+    return escn_file.add_external_resource(img_resource, image)
+
+
+# need to define in the start of fragment code
+FRAGMENT_OUT_COLOR = Variable('vec4', 'result_color')
+FRAGMENT_OUT_ETA = Variable('float', 'result_eta')
+FRAGMENT_OUT_TRANSMISSION = Variable('float', 'result_transmission')
+FRAGMENT_OUT_ANISOTROPY = Variable('float', 'result_anisotropy')
+
+FRAGMENT_OUT_METALLIC = Variable('float', 'METALLIC', True)
+FRAGMENT_OUT_SPECULAR = Variable('float', 'SPECULAR', True)
+FRAGMENT_OUT_ROUGHNESS = Variable('float', 'ROUGHNESS', True)
+FRAGMENT_OUT_CLEARCOAT = Variable('float', 'CLEARCOAT', True)
+FRAGMENT_OUT_CLEARCOAT_GLOSS = Variable('float', 'CLEARCOAT_GLOSS', True)
+FRAGMENT_OUT_SSS_STRENGTH = Variable('float', 'SSS_STRENGTH', True)
+FRAGMENT_OUT_EMISSION = Variable('vec3', 'EMISSION', True)
+FRAGMENT_OUT_NORMAL = Variable('vec3', 'NORMAL', True)
+FRAGMENT_OUT_TANGENT = Variable('vec3', 'TANGENT', True)
+
+FRAGMENT_OUT_BSDF = ShaderOutChannels()
+FRAGMENT_OUT_BSDF.set_channel('color', FRAGMENT_OUT_COLOR)
+FRAGMENT_OUT_BSDF.set_channel('sss_strength', FRAGMENT_OUT_SSS_STRENGTH)
+FRAGMENT_OUT_BSDF.set_channel('specular', FRAGMENT_OUT_SPECULAR)
+FRAGMENT_OUT_BSDF.set_channel('metallic', FRAGMENT_OUT_METALLIC)
+FRAGMENT_OUT_BSDF.set_channel('roughness', FRAGMENT_OUT_ROUGHNESS)
+FRAGMENT_OUT_BSDF.set_channel('clearcoat', FRAGMENT_OUT_CLEARCOAT)
+FRAGMENT_OUT_BSDF.set_channel('clearcoat_gloss', FRAGMENT_OUT_CLEARCOAT_GLOSS)
+FRAGMENT_OUT_BSDF.set_channel('anisotropy', FRAGMENT_OUT_ANISOTROPY)
+FRAGMENT_OUT_BSDF.set_channel('transmission', FRAGMENT_OUT_TRANSMISSION)
+FRAGMENT_OUT_BSDF.set_channel('eta', FRAGMENT_OUT_ETA)
+FRAGMENT_OUT_BSDF.set_channel('emission', FRAGMENT_OUT_EMISSION)
+FRAGMENT_OUT_BSDF.set_channel('normal', FRAGMENT_OUT_NORMAL)
+FRAGMENT_OUT_BSDF.set_channel('tangent', FRAGMENT_OUT_TANGENT)
+
+
+class MaterialScript:
+    # pylint: disable-msg=too-many-public-methods
```

Comment:
Hmm..... While pylint's limits are a bit low, they attempt to stop classes getting to "big". I wonder if this one could be split up somehow. Ideally you shouldn't suppress the warning.

Reply:
yes, I will solve it later, I had put a `FIXME` there